src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java

```diff
@@ -175,50 +175,53 @@ public ExpressionAtlasContentService(
+                        (BaselineExperiment) experiment, contentType);
+                otherExternalResourceLinks = externalResourceType.equals("geo") ?
+                        rnaSeqBaselineExperimentExternallyAvailableContentServiceGeo.list(
+                                (BaselineExperiment) experiment, contentType) :
+                        externalResourceType.equals("ega") ?
+                                rnaSeqBaselineExperimentExternallyAvailableContentServiceEga.list(
+                                        (BaselineExperiment) experiment, contentType) :
+                                rnaSeqBaselineExperimentExternallyAvailableContentServiceEna.list(
+                                        (BaselineExperiment) experiment, contentType);
+                arrayExpressLinks.addAll(otherExternalResourceLinks);
+                break;
+            case RNASEQ_MRNA_DIFFERENTIAL:
+                arrayExpressLinks = rnaSeqDifferentialExperimentExternallyAvailableContentService.list(
+                        (DifferentialExperiment) experiment, contentType);
+                otherExternalResourceLinks = externalResourceType.equals("geo") ?
+                        rnaSeqDifferentialExperimentExternallyAvailableContentServiceGeo.list(
+                                (DifferentialExperiment) experiment, contentType) :
+                        externalResourceType.equals("ega") ?
+                                rnaSeqDifferentialExperimentExternallyAvailableContentServiceEga.list(
+                                        (DifferentialExperiment) experiment, contentType) :
+                                rnaSeqDifferentialExperimentExternallyAvailableContentServiceEna.list(
+                                        (DifferentialExperiment) experiment, contentType);
+                arrayExpressLinks.addAll(otherExternalResourceLinks);
+                break;
+            default:
+                arrayExpressLinks = microarrayExperimentExternallyAvailableContentService.list(
+                        (MicroarrayExperiment) experiment, contentType);
+                otherExternalResourceLinks =  externalResourceType.equals("geo") ?
+                        microarrayExperimentExternallyAvailableContentServiceGeo.list(
+                                (MicroarrayExperiment) experiment, contentType) :
+                        externalResourceType.equals("ega") ?
+                                microarrayExperimentExternallyAvailableContentServiceEga.list(
+                                        (MicroarrayExperiment) experiment, contentType) :
+                                microarrayExperimentExternallyAvailableContentServiceEna.list(
+                                        (MicroarrayExperiment) experiment, contentType);
+                arrayExpressLinks.addAll(otherExternalResourceLinks);
         }
+        return arrayExpressLinks;
```

Comment:
Sorry to be picky, but at this point, `arrayExpressLinks` can be just AE links (e.g. if the experiment is of type proteomics baseline), but in all other cases it will contain other sort of links (GEO, EGA or ENA), so I think this is not a very good name.

Instead, use a builder as suggested above with a more generic name (e.g. `externalResourcesLinks`, `arrayExpressAndOtherExternalResourcesLinks`, whatever).